packages/block-library/src/embed/util.js

```diff
@@ -198,6 +198,14 @@ export function getClassNames(
 		) {
 			const potentialRatio = ASPECT_RATIOS[ ratioIndex ];
 			if ( aspectRatio >= potentialRatio.ratio ) {
+				// Evaluate the difference between actual aspect ratio and closest match.
+				// If the difference is too big, do not scale the embed according to aspect ratio.
+				const ratioDiff = aspectRatio - potentialRatio.ratio;
+				if ( ratioDiff > 0.1 ) {
+					// No close aspect ratio match found.
+					return removeAspectRatioClasses( existingClassNames );
```

Comment:
This seems incorrect, as it marks the post dirty, all the time.

Reply:
This already was the case in trunk for some other embeds (I've tested twitter and reddit, could be more). #13706 is an old (now conflicted) PR to fix it and the description there does nicely detail the cause. The changes needed to resolve it are pretty simple and I could push them if you'd like. It could also be tackled separately.

Reply:
Oh wow, great investigative work. I would be very grateful if you pushed the fixes!

Reply:
Should be resolved. I added a couple of unit tests to help keep it that way. There were some failures in CI from snapshot tests. I verified the changes are expected and updated the snapshots. I've also rebased this in hopes it would clear up some lint failures that had also come up 🤞 .